sw/airborne/modules/ins/ins_lpe.c

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright (C) 2016 Michal Podhradsky <michal.pohradsky@aggiemail.usu.edu>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file modules/ins/ins_lpe.c
+ *
+ * Local Position Estimator
+ */
+#include "modules/ins/ins_lpe.h"
+#include "subsystems/abi.h"
+#include "mcu_periph/sys_time.h"
+#include "message_pragmas.h"
+#include "subsystems/imu.h"
+
+#ifndef USE_INS_NAV_INIT
+#define USE_INS_NAV_INIT TRUE
+PRINT_CONFIG_MSG("USE_INS_NAV_INIT defaulting to TRUE")
+#endif
+
+#ifndef P_MAX // max allowed value in state covariance
+#define P_MAX 1.0e6
```

Comment:
for single precision numerical constant, it is usually better to specify it with `1.0e6f`, same for the rest of the code